### PR TITLE
Add identifier to NEXT_DATA for gs(s)p

### PR DESCRIPTION
**Pull request by ijjk**
This adds a `gsp` or `gssp` field to `__NEXT_DATA__` to allow detecting whether a page is using `getStaticProps` or `getServerSideProps`. I added tests to the respective test suites to ensure this is set correctly 

[NEXT-98]

[NEXT-98]: https://zeit.atlassian.net/browse/NEXT-98

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 8.7s | 8.9s | ⚠️ +127ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| index.html gzip | 936 B | 936 B | ✓ |
| link.html gzip | 946 B | 946 B | ✓ |
| withRouter.html gzip | 934 B | 934 B | ✓ |
| Overall change | 2.82 kB | 2.82 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 9.5s | 9.4s | -44ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall decrease ✓</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _error.js gzip | 292 kB | 292 kB | ⚠️ +468 B |
| 404.html gzip | 1.34 kB | 1.34 kB | ✓ |
| hooks.html gzip | 975 B | 975 B | ✓ |
| index.js gzip | 292 kB | 292 kB | -31 B |
| link.js gzip | 300 kB | 300 kB | ⚠️ +9 B |
| routerDirect.js gzip | 298 kB | 298 kB | ⚠️ +54 B |
| withRouter.js gzip | 299 kB | 298 kB | -1.02 kB |
| Overall change | 1.48 MB | 1.48 MB | -517 B |

</details>

</details>

File: test/integration/prerender/test/index.test.js [hunk 1]
```diff
@@ -282,6 +282,18 @@ const runTests = (dev = false, looseMode = false) => {
...
+  it('should have gsp in __NEXT_DATA__', async () => {
+    const html = await renderViaHTTP(appPort, '/')
+    const $ = cheerio.load(html)
+    expect(JSON.parse($('#__NEXT_DATA__').text()).gsp).toBe(true)
+  })
+
+  it('should not have gsp in __NEXT_DATA__ for non-GSP page', async () => {
+    const html = await renderViaHTTP(appPort, '/normal')
+    const $ = cheerio.load(html)
+    expect(JSON.parse($('#__NEXT_DATA__').text()).gsp).toBeFalsy()
```
Comment (Timer) [inline test/integration/prerender/test/index.test.js]:
```suggestion
    expect('gsp' in JSON.parse($('#__NEXT_DATA__').text())).toBe(false)
```

File: test/integration/getserversideprops/test/index.test.js [hunk 2]
```diff
@@ -191,6 +191,18 @@ const runTests = (dev = false) => {
...
+  it('should have gssp in __NEXT_DATA__', async () => {
+    const html = await renderViaHTTP(appPort, '/')
+    const $ = cheerio.load(html)
+    expect(JSON.parse($('#__NEXT_DATA__').text()).gssp).toBe(true)
+  })
+
+  it('should not have gssp in __NEXT_DATA__ for non-GSSP page', async () => {
+    const html = await renderViaHTTP(appPort, '/normal')
+    const $ = cheerio.load(html)
+    expect(JSON.parse($('#__NEXT_DATA__').text()).gssp).toBeFalsy()
```
Comment (Timer) [inline test/integration/getserversideprops/test/index.test.js]:
```suggestion
    expect('gssp' in JSON.parse($('#__NEXT_DATA__').text())).toBe(false);
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 8.7s | 8.7s | -71ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| index.html gzip | 936 B | 936 B | ✓ |
| link.html gzip | 946 B | 946 B | ✓ |
| withRouter.html gzip | 934 B | 934 B | ✓ |
| Overall change | 2.82 kB | 2.82 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 9.5s | 9.5s | ⚠️ +64ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _error.js gzip | 292 kB | 292 kB | ⚠️ +193 B |
| 404.html gzip | 1.34 kB | 1.34 kB | ✓ |
| hooks.html gzip | 975 B | 975 B | ✓ |
| index.js gzip | 292 kB | 293 kB | ⚠️ +1.18 kB |
| link.js gzip | 299 kB | 299 kB | ⚠️ +15 B |
| routerDirect.js gzip | 298 kB | 298 kB | -262 B |
| withRouter.js gzip | 298 kB | 298 kB | -22 B |
| Overall change | 1.48 MB | 1.48 MB | ⚠️ +1.1 kB |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 10.2s | 10.5s | ⚠️ +375ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| index.html gzip | 936 B | 936 B | ✓ |
| link.html gzip | 946 B | 946 B | ✓ |
| withRouter.html gzip | 934 B | 934 B | ✓ |
| Overall change | 2.82 kB | 2.82 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.8s | -92ms |
| nodeModulesSize | 56.3 MB | 56.3 MB | ⚠️ +483 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.76 kB | 5.76 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..67cb.js gzip | 9.69 kB | 9.69 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 55.3 kB | 55.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.64 kB | 6.64 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.3 kB | 51.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 18.9 kB | 18.9 kB | ✓ |
| Overall change | 18.9 kB | 18.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.js gzip | 1.07 kB | 1.07 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.38 kB | 7.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _app.module.js gzip | 589 B | 589 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.48 kB | 1.48 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.27 kB | 5.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall decrease ✓</summary>

|  | zeit/next.js canary  | ijjk/next.js add/gssp-identifier | Change |
| - | - | - | - |
| _error.js gzip | 292 kB | 292 kB | -539 B |
| 404.html gzip | 1.34 kB | 1.34 kB | ✓ |
| hooks.html gzip | 975 B | 975 B | ✓ |
| index.js gzip | 292 kB | 292 kB | -82 B |
| link.js gzip | 299 kB | 299 kB | ⚠️ +79 B |
| routerDirect.js gzip | 298 kB | 299 kB | ⚠️ +960 B |
| withRouter.js gzip | 299 kB | 298 kB | -1.02 kB |
| Overall change | 1.48 MB | 1.48 MB | -600 B |

</details>

</details>

**Pull request merged**